server/sttp-mock-server/src/main/scala/sttp/tapir/server/mockserver/SttpMockServerClient.scala

```diff
@@ -0,0 +1,78 @@
+package sttp.tapir.server.mockserver
+
+import io.circe.Printer
+import io.circe.syntax._
+import sttp.client3._
+import sttp.client3.testing._
+import sttp.model.Uri.UriContext
+import sttp.model.{MediaType, StatusCode, Uri}
+import sttp.tapir.client.sttp.SttpClientInterpreter
+import sttp.tapir.internal.{NoStreams, ParamsAsAny}
+import sttp.tapir.server.internal.{EncodeOutputBody, EncodeOutputs, OutputValues}
+import sttp.tapir.{CodecFormat, Endpoint, RawBodyType, WebSocketBodyOutput}
+import sttp.tapir.server.mockserver.impl.JsonCodecs._
+import java.nio.charset.Charset
+import io.circe.parser._
+
+class SttpMockServerClient[F[_]](baseUri: Uri)(backend: SttpBackend[F, Any]) {
+  private val F = backend.responseMonad
+
+  def createExpectation[E, I, O](
+      endpoint: Endpoint[I, E, O, Any]
+  )(input: I, expectedOutput: Either[E, O], statusCode: StatusCode): F[List[Expectation]] = {
+    val request = SttpClientInterpreter.toRequest(endpoint, None).apply(input)
+
+    val requestBody = Option(new String(request.forceBodyAsByteArray)).filter(_.nonEmpty)
+
+    val response = {
+      val responseValue = expectedOutput.merge
+      val encodeOutputBody: EncodeOutputBody[Any, Any, Nothing] = new EncodeOutputBody[Any, Any, Nothing] {
```

Comment:
Probably can be refactored so that both sttp-stub-server and sttp-mock-server will use the same logic